AnomalyDetection/RandomCutForest/src/main/java/com/amazonaws/services/kinesisanalytics/FlinkStreamingJob.java

```diff
@@ -0,0 +1,65 @@
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.connectors.kinesis.FlinkKinesisConsumer;
+import org.apache.flink.streaming.connectors.kinesis.config.AWSConfigConstants;
+import org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants;
+
+import java.util.Properties;
+
+public class FlinkStreamingJob {
+    private static DataStream<InputData> createSource(
+            final StreamExecutionEnvironment env,
+            final String inputStreamName,
+            final String region) {
+        Properties inputProperties = new Properties();
+        inputProperties.setProperty(ConsumerConfigConstants.AWS_REGION, region);
+
+        return env.addSource(new FlinkKinesisConsumer<>(inputStreamName, new DataSerializationSchema(), inputProperties));
+    }
+
+    private static KinesisStreamsSink<OutputData> createSink(final String outputStreamName, final String region) {
+        Properties outputProperties = new Properties();
+        outputProperties.setProperty(AWSConfigConstants.AWS_REGION, region);
+
+        return KinesisStreamsSink.<OutputData>builder()
+                .setKinesisClientProperties(outputProperties)
+                .setSerializationSchema(new DataSerializationSchema())
+                .setStreamName(outputProperties.getProperty("OUTPUT_STREAM", outputStreamName))
+                .setPartitionKeyGenerator(element -> String.valueOf(element.hashCode()))
+                .build();
+    }
+
+    public static void main(String[] args) throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+        Properties applicationProperties = KinesisAnalyticsRuntime.getApplicationProperties().get("RcfExampleEnvironment");
+        String region = applicationProperties.getProperty("region", "us-west-2");
+        String inputStreamName = applicationProperties.getProperty("inputStreamName", "ExampleInputStream-RCF");
+        String outputStreamName = applicationProperties.getProperty("outputStreamName", "ExampleOutputStream-RCF");
+
+        DataStream<InputData> source = createSource(env, inputStreamName, region);
+
+        RandomCutForestOperator<InputData, OutputData> randomCutForestOperator =
+                RandomCutForestOperator.<InputData, OutputData>builder()
+                        .setDimensions(1)
+                        .setShingleSize(1)
+                        .setSampleSize(628)
+                        .setInputDataMapper((inputData) -> new float[]{inputData.getValue()})
+                        .setResultMapper(((inputData, score) -> new OutputData(inputData.getTime(), inputData.getValue(), score)))
+                        .build();
+
+        source
+                .process(randomCutForestOperator, TypeInformation.of(OutputData.class)).setParallelism(1)
```

Comment:
Parallelism for this operator is explicitly limited to 1 to ensure correct behaviour.
Since anomaly detection algorithm is trained online, running it in parallel on multiple taskmanagers can produce unexpected results depending on distribution of data.